numpy/core/tests/test_multiarray.py

```diff
@@ -4643,6 +4643,12 @@ def test_overlaps(self):
         np.putmask(x[1:4], x[:3], [True, False, True])
         assert_equal(x, np.array([True, True, True, True]))
 
+    def test_writeable(self):
+        a = np.arange(12)
+        a.setflags(write = False)
+
+        with pytest.raises(Exception) as e_info:
```

Comment:
It would be nicer if you make this specific, i.e. not `Exception` but what ever it is (probably a `TypeError`). You also do not use `e_info`, so just remove it.  If you want to be precise, you can add `match="putmask: output array"` which will check the error message as well.